docs/source/data.extend.rst

```diff
@@ -16,3 +16,22 @@ Moreover, it is designed to be extensible and users can bring their own data per
     :no-members:
     :no-inherited-members:
     :no-special-members:
+
+The ``fsspec`` Data Plugin
+--------------------------
+
+Flytekit ships with a default storage driver that uses aws-cli on AWS and gsutil on GCP. By default, Flyte uploads the task outputs to S3 or GCS using these storage drivers.
+
+Why ``fsspec``?
+^^^^^^^^^^^^^^^
+
+You can use the fsspec plugin implementation to utilize all its available plugins with flytekit. The `fsspec <https://github.com/flyteorg/flytekit/blob/aad4e0e67fbc1614b4aa3e49c49cd777738cd3d7/plugins/flytekit-data-fsspec/setup.py#L25-L26>`_ plugin provides an implementation of the data persistence layer in Flytekit. For example: HDFS, FTP are supported in fsspec, so you can use them with flytekit too.
+The data persistence layer helps store logs of metadata and raw user data.
+As a consequence of the implementation, an S3 driver can be installed using ``pip install s3transfer``.
```

Comment:
I think it should be `s3fs`, https://filesystem-spec.readthedocs.io/en/latest/api.html?highlight=s3fs#other-known-implementations
```suggestion
As a consequence of the implementation, an S3 driver can be installed using ``pip install s3fs``.
```

Reply:
Sure, will fix that.